src/core/agent/services/connectionService.ts

```diff
@@ -129,10 +147,21 @@ class ConnectionService extends AgentService {
           connection,
         };
       }
-    }
-    await this.createConnectionMetadata(connectionId, connectionMetadata);
+      await this.createConnectionMetadata(connectionId, connectionMetadata);
+    } else {
+      // @TODO - foconnor: We shouldn't emit this if it's a multiSigInvite, but the routing will break if we don't.
+      // To fix once we handle errors for the scanner in general.
+      this.props.eventEmitter.emit<ConnectionStateChangedEvent>({
+        type: EventTypes.ConnectionStateChanged,
+        payload: {
+          isMultiSigInvite: multiSigInvite,
+          connectionId: undefined,
+          status: ConnectionStatus.PENDING,
+          url,
+        },
+      });
 
-    if (!multiSigInvite) {
+      await this.createConnectionMetadata(connectionId, connectionMetadata);
       this.props.eventEmitter.emit<ConnectionStateChangedEvent>({
```

Comment:
We should only be emitting this for multi-sig, so this should be in the other branch of the if statement.

This branch is for a normal connection - we just emitted that it was pending, we can't just emit that it's confirmed straight away. That's the job of the long running operation tracker.

Reply:
This isn't resolved...